test/vinyl-luatest/update_optimize_test.lua

```diff
@@ -0,0 +1,530 @@
+local t = require('luatest')
+
+local server = require('test.luatest_helpers.server')
+
+local g = t.group()
+
+local dump_stmt_count_str = [[function dump_stmt_count(indexes)
+    local dumped_count = 0
+    for _, i in ipairs(indexes) do
+        dumped_count = dumped_count +
+            box.space.test.index[i]:stat().disk.dump.output.rows
+    end
+    return dumped_count
+end]]
+
+g.before_all(function()
+    local box_cfg = require('test.vinyl-luatest.box_cfg')
+    g.server = server:new({alias = 'master', box_cfg = box_cfg})
+    g.server:start()
+    g.server:eval(dump_stmt_count_str)
+end)
+
+g.before_each(function()
+    g.server:exec(function()
+        box.schema.space.create('test', {engine = 'vinyl'})
+    end)
+end)
+
+g.after_each(function()
+    g.server:exec(function() box.space.test:drop() end)
+end)
+
+g.test_optimize_one_index = function()
+    g.server:exec(function()
+        local t = require('luatest')
+
+        box.space.test:create_index('primary', {run_count_per_level = 20})
+        box.space.test:create_index('secondary',
+            {parts = {5, 'unsigned'}, run_count_per_level = 20})
+
+        box.snapshot()
+
+        local old_stmt_count = dump_stmt_count({'primary', 'secondary'})
```

Comment:
Please access globals via `_G` and drop the luacheck exception:

```lua
local old_stmt_count = _G.dump_stmt_count({'primary', 'secondary'})
```

Reply:
Fixed.